specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2022-10-01-preview/ResourceHealth.json

```diff
@@ -0,0 +1,2155 @@
+          "description": "It provides the Timestamp for the given update for the service health event.",
+          "format": "date-time"
+        }
+      }
+    },
+    "faq": {
+      "type": "object",
+      "description": "Frequently asked question for the service health event",
+      "properties": {
+        "question": {
+          "type": "string",
+          "description": "FAQ question for the service health event."
+        },
+        "answer": {
+          "type": "string",
+          "description": "FAQ answer for the service health event."
+        },
+        "localeCode": {
+          "type": "string",
+          "description": "FAQ locale for the service health event."
+        }
+      }
+    },
+    "impactedResourceStatus": {
+      "type": "object",
+      "description": "impactedResource with health status",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "description": "Properties of impacted resource status.",
+          "properties": {
+            "availabilityState": {
+              "type": "string",
+              "description": "Impacted resource status of the resource.",
+              "enum": [
+                "Available",
+                "Unavailable",
+                "Degraded",
+                "Unknown"
+              ],
+              "x-ms-enum": {
+                "name": "AvailabilityStateValues",
+                "modelAsString": true
+              }
+            },
+            "title": {
+              "type": "string",
+              "description": "Title description of the impacted resource status."
+            },
+            "summary": {
+              "type": "string",
+              "description": "Summary description of the impacted resource status."
+            },
+            "reasonType": {
+              "type": "string",
+              "description": "When the resource's availabilityState is Unavailable, it describes where the health impacting event was originated.",
+              "enum": [
+                "Unplanned",
+                "Planned",
+                "UserInitiated"
+              ],
+              "x-ms-enum": {
+                "name": "ReasonTypeValues",
+                "modelAsString": true
+              }
+            },
+            "occurredTime": {
+              "type": "string",
+              "description": "Timestamp for when last change in health status occurred.",
+              "format": "date-time"
+            }
+          }
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "MetadataEntityListResult": {
+      "description": "The list of metadata entities",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of metadata entities.",
+          "type": "array",
```

Comment:
>array [](http://example.com/codeflow?start=19&length=5)

use x-ms-identifiers for the object in the array: [https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-identifiers](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-identifiers)

Reply:
so according to your link:
"The default identifying property is id. You don't need to add this annotataion if there is the id property in the object model."
since metadata entity has the id field via:
"allOf": [
        {
          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
        }
      ]
 we don't need to add it

Reply:
Sounds good, I missed this in this initial review